gflownet/gflownet.py

```diff
@@ -475,12 +460,27 @@ def sample_batch(
                 dict_replay = pickle.load(f)
                 n_replay = min(n_replay, len(dict_replay["x"]))
                 envs = [self.env.copy().reset(idx) for idx in range(n_replay)]
-                # TODO: implement other sampling options besides permutation
                 if n_replay > 0:
                     x_replay = [x for x in dict_replay["x"].values()]
-                    x_replay = [x_replay[idx] for idx in self.rng.permutation(n_replay)]
-            actions = []
-            valids = []
+                    if self.replay_sampling == "permutation":
+                        x_replay = [
+                            x_replay[idx] for idx in self.rng.permutation(n_replay)
+                        ]
+                    elif self.replay_sampling == "weighted":
+                        x_rewards = np.array(
+                            [r for r in dict_replay["rewards"].values()]
+                        )
```

Comment:
Convert to numpy array through `np.fromiter` instead of through iteration (a bit faster and more readable).
```suggestion
                        x_rewards = np.fromiter(dict_replay["rewards"].values(), dtype=float)
```